core/src/main/scala/kafka/utils/ZkUtils.scala

```diff
@@ -795,22 +795,9 @@ class ZkUtils(val zkClient: ZkClient,
     * client updates the path stat.version gets incremented
     */
   def getSequenceId(path: String, acls: java.util.List[ACL] = DefaultAcls): Int = {
-    try {
+      makeSurePersistentPathExists(path)
```

Comment:
The disadvantage of this call to make sure the znode exists is that we're making an additional exists call every time we call `getSequenceId`. It isn't a big deal because we don't call this often, so I don't have a super strong preference. It also works for me if you want to leave the catch block below and return `stat.getVersion + 1` at line 800.